### PR TITLE
feat: add prefill/decode per-pool GPU budget for hetero disagg

### PR DESCRIPTION
**Pull request by mandy-li**
Introduce max_prefill_gpus and max_decode_gpus parameters to allow independent GPU limits for prefill and decode pools in disagg serving. This enables asymmetric resource allocation (e.g., 4 prefill GPUs + 12 decode GPUs) that total_gpus alone cannot express.

- Add max_prefill_gpus/max_decode_gpus to find_best_disagg_result_under_constraints
- Plumb through pareto_analysis.disagg_pareto and TaskRunner.run_disagg
- Add defaults (None) in TaskConfigFactory replica_config
- Validate that values are positive when set (reject 0 or negative)
- Add TestHeteroDisaggGPUBudget test class with 7 test cases

## Release Notes

* **New Features**
  * Added optional GPU budget constraints for prefill and decode stages, enabling fine-grained control over disaggregated inference worker selection.
  * Input validation rejects invalid budget values.

* **Tests**
  * Comprehensive test coverage for GPU budget constraints, including symmetric and asymmetric scenarios and edge cases.

### CODE REVIEW

File: src/aiconfigurator/sdk/task.py [hunk 1]
```diff
@@ -1354,6 +1356,8 @@ def _wc_get(wc: object, key: str, fallback):
             max_num_gpu=task_config.replica_config.max_gpu_per_replica,
             prefill_max_num_worker=task_config.replica_config.max_prefill_worker,
             decode_max_num_worker=task_config.replica_config.max_decode_worker,
+            max_prefill_gpus=task_config.replica_config.get("max_prefill_gpus"),
+            max_decode_gpus=task_config.replica_config.get("max_decode_gpus"),
```
Comment (Arsene12358) [inline src/aiconfigurator/sdk/task.py:1360]: nitpick, can we access max_prefill_gpus and max_decode_gpus via dot notation instead ? to be consistent with the surrounding codes

**Review by tianhaox — APPROVED** [review]

File: src/aiconfigurator/sdk/inference_session.py [hunk 2]
```diff
@@ -558,6 +565,13 @@ def _match_workers(
                     if len(num_gpu_set) > 0 and num_gpu not in num_gpu_set:
                         continue
 
+                    # per-pool GPU budget for hetero disagg
+                    if max_prefill_gpus is not None and max_decode_gpus is not None:
+                        if prefill_gpus * prefill_num_worker > max_prefill_gpus:
+                            continue
+                        if decode_gpus * decode_num_worker > max_decode_gpus:
+                            continue
+
```
Comment (Arsene12358) [inline src/aiconfigurator/sdk/inference_session.py:574]: I think it would be better to apply filter independently. In the current approach, if one of max_prefill_gpus or max_decode_gpus is none, the filter is skipped entirely
```python
if max_prefill_gpus is not None and prefill_gpus * prefill_num_worker > max_prefill_gpus:
    continue
if max_decode_gpus is not None and decode_gpus * decode_num_worker > max_decode_gpus:
    continue
```
